src/core/services/NetworkCommunicatorService.ts

```diff
@@ -0,0 +1,159 @@
+   */
+  private verify(authHeader?: string) {
+    const token = extractToken(authHeader);
+    if (!token) {
+      throw new JsonWebTokenError("No token found");
+    }
+    return verify(token, config.secret);
+  }
+
+  // Event methods
+  public on = this.events.on;
+  public once = this.events.once;
+  public off = this.events.off;
+
+  /**
+   * Start the internal server
+   */
+  public startServer() {
+    const internalApp = express();
+    this.server = internalApp.listen(4000);
+
+    process.on("SIGTERM", () => {
+      this.server?.close();
+    });
+
+    // Register internal service routes
+    internalApp.post(
+      "/reload",
+      wrapAsync(this.onInternalServiceRequest.bind(this))
+    );
+
+    return internalApp;
+  }
+
+  /**
+   * Called if this service is notified from other internal service to a registered route
+   * * Verifies the request token
+   * * If the request is authenticated, it will emit the event with the payload
+   * @throws JsonWebTokenError if the request is not authenticated
+   * @emits The event with the payload, e.g. "user:created" if the route is "/user/created" or 'reload' if the route is '/reload'
+   * @param req
+   * @param res
+   */
+  private async onInternalServiceRequest(req: Request, res: Response) {
+    // Get the action from request path
+    const actionPath = req.path.substring(1);
+    // Convert action path to event name
+    const eventName = actionPath.replaceAll("/", ":");
+    try {
+      const payload = this.verify(req.headers?.authorization);
+      this.events.emit(eventName, payload);
+      return res.sendStatus(200);
+    } catch (error) {
+      if (error instanceof JsonWebTokenError) {
+        // Not authenticated internal service request
+        return res.sendStatus(401);
+      } else {
+        throw error;
+      }
+    }
+  }
+
+  /**
+   * Notify an internal service by `serviceName`
+   * @param serviceName The name of the service
+   * @param actionPath The path of the action
+   * @param data The data to send
+   * @returns
+   */
+  private async notify(
+    serviceName: string,
+    actionPath: string,
+    payload: string | Buffer | object = {}
+  ) {
+    const service = this.services[serviceName];
+    if (!service) {
+      throw new Error(`Internal service "${serviceName}" not found`);
+    }
+
+    const headers = {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${this.sign(payload)}`
+    };
+
+    try {
+      // Payload parameter is undefined here because the payload is encrypted in the bearer token
+      return await axios.post(
+        `http://${serviceName}:4000/${actionPath}`,
+        undefined,
+        {
+          headers
+        }
+      );
+    } catch (error) {
+      // If the service is optional and the request fails, ignore the error otherwise log it
+      if (!service.optional) {
+        log.error("Failed to notify webhook of options change", error);
```

Comment:
Just a tiny thing and it's was a problem in my old code too, but could you change this to log which service has failed instead of "webhook"